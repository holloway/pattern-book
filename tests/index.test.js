import React from "react";
import Book from "../src/index";
import renderer from "react-test-renderer";

it("renders correctly", async () => {
  global.document = {
    styleSheets: [{ childNodes: ["b", "c"] }]
  };

  const book = (
    <Book>
      <h1>test</h1>
    </Book>
  );

  const createNodeMock = elem => {
    return {
      innerHTML: "<h1>test</h1>",
      childNodes: ["a", "b"]
    };
  };
  const tree = renderer.create(book, { createNodeMock });
  const instance = tree.getInstance();

  expect(tree.toJSON()).toMatchSnapshot();
});
