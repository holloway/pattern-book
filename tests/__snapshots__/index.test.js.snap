// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pattern-book"
>
  <div
    className="pattern-book__example"
    onClick={[Function]}
  >
    <h1>
      test
    </h1>
  </div>
  <details
    className="pattern-book__html"
  >
    <summary>
      HTML
    </summary>
    <code
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span class=\\"b-tag-start\\">&lt;</span><span class=\\"b-tag\\">h1</span><span class=\\"b-tag-end\\">&gt;</span><br/>
 &nbsp;<span class=\\"b-text\\">test</span><br/>
<span class=\\"b-tag-start\\">&lt;</span><span class=\\"b-tag-end\\">/</span><span class=\\"b-tag\\">h1</span><span class=\\"b-tag-end\\">&gt;</span><br/>
",
        }
      }
    />
  </details>
  <details
    className="pattern-book__css"
  >
    <summary>
      CSS
    </summary>
    <code>
      
    </code>
  </details>
</div>
`;
